sdk/python/feast/feature_store.py

```diff
@@ -72,15 +73,31 @@ def __init__(
                 ),
             )
 
+        metadata_store_config = self.config.get_metadata_store_config()
+        self._registry = Registry(
+            registry_path=metadata_store_config.path,
+            cache_ttl=timedelta(seconds=metadata_store_config.cache_ttl_seconds),
+        )
+
     @property
     def project(self) -> str:
         return self.config.project
 
     def _get_provider(self) -> Provider:
         return get_provider(self.config)
 
-    def _get_registry(self) -> Registry:
-        return Registry(self.config.metadata_store)
+    def refresh_registry(self):
+        """Fetches and caches a copy of the feature registry in memory.
+
+        Explicitly calling this method makes it possible for methods like get_online_features to use cached registry
+        state instead of retrieving the registry state at request-time, thus reducing latency."""
```

Comment:
can we clarify a little more how this works, i.e. if this function is never called then every cache_ttl seconds the get_online_features request will refresh the cache but if this function is called it refreshes the cache and resets the timer back to cache_ttl seconds